views/pages/tasks.ejs

```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <%- include('../partials/head'); %>
+</head>
+<body class="container">
+
+<header>
+  <%- include('../partials/header'); %>
+</header>
+
+<main>
+  <p><%= message %></p>
+  <div class="jumbotron">
+    <% if (tasks.length > 0) { %>
+      <table class="table"
+        <tr><th>Name</th><th>Completed</th><th colspan="2"></th></tr>
+        <% tasks.forEach(function(task){ %>
+          <tr><td><%= task.name %></td><td><%= task.completed %></td>
+            <td><a href=<%= "/tasks/edit/" + task.id %>  class="btn btn-primary">Edit</a></td>
```

Comment:
Regarding the other comment about putting `task.completed` directly into the `value` field of the checkbox, it looks like this pattern is useful for getting that done! Here is an example of using `ejs` to put a value right into the `href` field. So, I think that in that other case, you can do something like this


```
<input type="checkbox" value=<%= task.checked.toString() %>
```

And you can experiment with whether the `.toString()` call is necessary; I expect it might not be.

Reply:
I tried this. But the checked attribute of the input checkbox is not checked on the edit page even though the task is completed cos now in the input tag we are missing the checked attribute.

Reply:
.toString() is not required and it throws error as task.checked is undefined.

Reply:
Ok, can we think of an expression that works, considering that `task.completed` is sometimes undefined?